gradio/components/audio.py

```diff
@@ -18,11 +19,13 @@
 set_documentation_group("component")
 
 
-class WaveformOptions(TypedDict, total=False):
-    waveform_color: str
-    waveform_progress_color: str
-    show_controls: bool
-    skip_length: int
+@dataclasses.dataclass
+class WaveformOptions:
+    waveform_color: Optional[str] = None
+    waveform_progress_color: Optional[str] = None
+    show_controls: Optional[bool] = False
+    skip_length: Optional[int] = None
+    show_recording_waveform: Optional[bool] = True
```

Comment:
Nit: we usually adopt this style in the codebaser:
```suggestion
    waveform_color: str | None = None
    waveform_progress_color: str | None = None
    show_controls: bool | None = False
    skip_length: int | None = None
    show_recording_waveform: bool | None = True
```

Reply:
Followup: should the two `bool` parameters actually be optional?

Reply:
Ah sorry just to clarify, I mean that we should do something like this:

```py
    waveform_color: str | None = None
    waveform_progress_color: str | None = None
    show_controls: bool = False
    skip_length: int | None = None
    show_recording_waveform: bool = True
```

So they are indeed optional and if they are not provided, they take the default values, but they do not take `None` as an option